stacks-signer/src/stacks_client.rs

```diff
@@ -54,6 +70,34 @@ impl From<&Config> for StacksClient {
     }
 }
 
+/// Trait used to make interact with Clarity contracts for use in the signing process
+pub trait InteractWithStacksContracts {
```

Comment:
Really nitpicky. Can ignore if you like, but think maybe something like "StacksContractCallable" is better. I can be easily convinced otherwise though :P

Reply:
Can I ask what the trait definition here is for?  Are there two or more implementations?  If not, then perhaps the trait can be dropped altogether.

Reply:
A lot of the trait's functions take the same arguments over and over (e.g. `http_origin`, `chain_id`, `tx_version`).  I think these are values that the trait implementation can just supply.  Why would a caller ever need to supply different values for the same trait implementation instantiation?